### PR TITLE
Add error for no parameters on jsonprofilemerger

### PR DESCRIPTION
**Pull request by Qinusty**
**What this PR does and why we need it:**

It might save someone some time looking at the stack trace.

**New changes / Issues that this PR fixes:**

jsonprofilemerger did not provide any check for an empty parameter list, but errored further into execution when none were provided.

**Special notes for reviewer:**

The new output is still not as clean as I'd like, but I wasn't sure the best way to handle reporting this to the user.

Example new output:
```
Traceback (most recent call last):
  File "/home/user/.cache/bazel/_bazel_user/c0fe45a85c85ea43d6d62cbc57b2f48c/execroot/__main__/bazel-out/k8-py2-fastbuild/bin/utils/json_profiles_merger.runfiles/__main__/utils/json_profiles_merger.py", line 70, in <module>
    app.run(main)
  File "/home/user/.cache/bazel/_bazel_user/c0fe45a85c85ea43d6d62cbc57b2f48c/execroot/__main__/bazel-out/k8-py2-fastbuild/bin/utils/json_profiles_merger.runfiles/third_party_pypi__absl_py_0_7_0/absl/app.py", line 300, in run
    _run_main(main, args)
  File "/home/user/.cache/bazel/_bazel_user/c0fe45a85c85ea43d6d62cbc57b2f48c/execroot/__main__/bazel-out/k8-py2-fastbuild/bin/utils/json_profiles_merger.runfiles/third_party_pypi__absl_py_0_7_0/absl/app.py", line 251, in _run_main
    sys.exit(main(argv))
  File "/home/user/.cache/bazel/_bazel_user/c0fe45a85c85ea43d6d62cbc57b2f48c/execroot/__main__/bazel-out/k8-py2-fastbuild/bin/utils/json_profiles_merger.runfiles/__main__/utils/json_profiles_merger.py", line 46, in main
    raise RuntimeError("At least one profile must be provided!")
RuntimeError: At least one profile must be provided!
```

**Does this require a change in the script's interface or the BigQuery's table structure?**

No

### CODE REVIEW

**Comment by Qinusty** [comment]
Added two commits fixing other cases where failures were noticed and could be avoided.

File: utils/json_profiles_merger.py [hunk 1]
```diff
@@ -42,6 +43,9 @@ def main(argv):
   # Discard the first argument (the binary).
   input_profiles = argv[1:]
 
+  if not input_profiles:
+      raise RuntimeError("At least one profile must be provided!")
```
Comment (joeleba) [inline utils/json_profiles_merger.py]: This should be `ValueError` instead, to be consistent with the rest of the repository. Also please use 2 spaces for indentation.

File: utils/json_profiles_merger_lib.py [hunk 2]
```diff
@@ -41,7 +41,7 @@ def write_to_csv(
     output_csv_path: a path to the output CSV file.
   """
   output_dir = os.path.dirname(output_csv_path)
-  if not os.path.exists(output_dir):
+  if output_dir and not os.path.exists(output_dir):
```
Comment (joeleba) [inline utils/json_profiles_merger_lib.py]: This is probably excessive. If `not output_dir` then `os.path.exists(output_dir)` would return False.

Comment (Qinusty) [reply utils/json_profiles_merger_lib.py]: This is required to check if the output_csv_path is a relative path to a file e.g. "output.csv"

```
output_csv_path = "output.csv"

output_dir = os.path.dirname(output_csv_path)
# output_dir = ""

not os.path.exists(output_dir)
# True

os.makedirs(output_dir)
# Stack trace...
# OSError: [Errno 2] No such file or directory: ''
```

File: utils/json_profiles_merger.py (hunk 1 above)
Comment (Qinusty) [reply utils/json_profiles_merger.py]: Good point, fixed :+1:

File: utils/json_profiles_merger.py [hunk 3]
```diff
@@ -18,6 +18,7 @@
 
 FLAGS = flags.FLAGS
 flags.DEFINE_string('output_path', None, 'The path to the output file.')
+flags.mark_flag_as_required("output_path")
```
Comment (joeleba) [inline utils/json_profiles_merger.py]: One final nit: please use single quote to be consistent with the rest of the file.
(should have mentioned this in my first review. Sorry about that.)

File: utils/json_profiles_merger.py [hunk 4]
```diff
@@ -42,6 +43,9 @@ def main(argv):
   # Discard the first argument (the binary).
   input_profiles = argv[1:]
 
+  if not input_profiles:
+    raise ValueError("At least one profile must be provided!")
```
Comment (joeleba) [inline utils/json_profiles_merger.py]: Same here.

**Pull request merged**